spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -88,22 +90,16 @@ public URI getURI() throws IOException {
 	}
 
 	private Blob resolve() throws IOException {
-		synchronized (this.monitor) {
-			if (this.blob == null) {
```

Comment:
Sorry, I forgot to explain that in the commit. Basically, I think of the `Resource` object as simply a light reference to the remote resource. Since this remote resource can be modified from anywhere, we can't really control the consistency of our cache.